frontend/src/components/contracts/ContractDetails.tsx

```diff
@@ -0,0 +1,155 @@
+import moment from "moment";
+
+import React from "react";
+
+import { Row, Col } from "react-bootstrap";
+
+import ContractsApi, * as C from "../../libraries/explorer-wamp/contracts";
+
+import CardCell from "../utils/CardCell";
+import TransactionLink from "../utils/TransactionLink";
+
+interface Props {
+  accountId: string;
+}
+
+interface State {
+  display: boolean;
+  locked: boolean;
+  contractInfo: C.ContractInfo;
+}
+
+export default class extends React.Component<Props, State> {
+  state: State = {
+    display: false,
+    locked: true,
+    contractInfo: {
```

Comment:
I guess that if you allow `contractInfo` to be `undefined` (which makes a lot more sense then this blank object), you won't need `display` flag as well since you will be able to check if there is anything in `contactInfo` in the `render`.